### ISSUE TITLE
Will this allow arbitrary/non-dictionary mnemonic seeds?

### ISSUE THREAD

**Issue by amritabithi** [mcdallas/cryptotools#39]
Hello,

I am trying to seed a private key with a string of random letters, and ran in to the error of  "cryptotools.BTC.HD.mnemonic.InvalidMnemonic"  trying to create one with an old key, becuse my words are not dictionary words included in the BIP.  Is there a way to initialize a private key with random text?

Thanks!

**Comment by mcdallas** [comment]
Hi @amritabithi ,

Do you want to create a regular private key or an extended key from your own phrase? Either way that’s really not recommended for a bunch of reasons:
1) if it’s not a standardized scheme you won’t be able to restore your wallet on any other software
2) you are very likely to not choose random words but instead choose low entropy memorable words that will lead to your coins being stolen. If you are going to choose random words just use the existing scheme

if you still want to do it you could devise a custom scheme to turn your string to bytes by passing it through sha256 for example (known as brainwallets which again you should not use)

```python
my_seed = sha256(str_to_bytes(“my custom words”))
```

and then for an extended key use `Xprv.from_seed(my_seed)` or for a regular key use `PrivateKey(my_seed)`

tl;dr don’t do it

**Comment by amritabithi** [comment]
Thank you for the quick response :)

What I am trying to do specifically is to generate the xpub from an old private key that I used which created wallets programmatically, and so I was hoping to use the xpub to scan the wallet and make sure it is in fact empty.

I never could get the "to_xpub()" function to work today, trying with the sample code, I could not seem to get python to recognize that the objects had any functions that I Imported, and eventually gave up.  I have been trying to find a script to show me an xpub for more than a year now.
